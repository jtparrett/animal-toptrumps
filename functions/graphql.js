const { ApolloServer } = require("apollo-server-lambda");
const faunadb = require("faunadb");
const q = faunadb.query;

const { FAUNADB_SECRET } = process.env;
const resolvers = require("./resolvers");
const typeDefs = require("./schema");

if (!FAUNADB_SECRET) {
  throw new Error("No FAUNADB_SECRET found.");
}

const server = new ApolloServer({
  typeDefs,
  resolvers,
  context: () => {
    const client = new faunadb.Client({ secret: FAUNADB_SECRET });

    return {
      client,
      q,
    };
  },
});

exports.handler = server.createHandler({
  cors: {
    origin: "*",
    credentials: true,
  },
});
