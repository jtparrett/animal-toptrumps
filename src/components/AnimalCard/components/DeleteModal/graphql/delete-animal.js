import { gql } from "@apollo/client";

export const DELETE_ANIMAL = gql`
  mutation($id: ID!) {
    deleteAnimal(id: $id)
  }
`;
