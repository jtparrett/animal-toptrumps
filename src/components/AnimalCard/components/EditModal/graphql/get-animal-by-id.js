import { gql } from "@apollo/client";

export const GET_ANIMAL_BY_ID = gql`
  query($id: ID!) {
    getAnimalById(id: $id) {
      id
      name
      type
      diet
      extinct
    }
  }
`;
