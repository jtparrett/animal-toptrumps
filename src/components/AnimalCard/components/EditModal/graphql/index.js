export * from "./get-animal-by-id";
export * from "./update-animal";
