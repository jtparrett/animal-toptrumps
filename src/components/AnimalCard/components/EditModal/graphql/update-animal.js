import { gql } from "@apollo/client";

export const UPDATE_ANIMAL = gql`
  mutation($id: ID!, $values: AnimalInput!) {
    updateAnimal(id: $id, values: $values) {
      id
      name
      type
      diet
      extinct
    }
  }
`;
