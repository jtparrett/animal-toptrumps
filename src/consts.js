import {
  GiBearFace,
  GiSnake,
  GiTropicalFish,
  GiFrog,
  GiSteak,
  GiHerbsBundle,
} from "react-icons/gi";

export const ANIMAL_TYPE = {
  MAMMAL: "MAMMAL",
  REPTILE: "REPTILE",
  FISH: "FISH",
  AMPHIBIOUS: "AMPHIBIOUS",
};

export const ANIMAL_DIET_TYPE = {
  CARNIVORE: "CARNIVORE",
  HERBIVORE: "HERBIVORE",
};

export const ANIMAL_TYPE_ICONS = {
  MAMMAL: GiBearFace,
  REPTILE: GiSnake,
  FISH: GiTropicalFish,
  AMPHIBIOUS: GiFrog,
};

export const ANIMAL_DIET_TYPE_ICONS = {
  CARNIVORE: GiSteak,
  HERBIVORE: GiHerbsBundle,
};
