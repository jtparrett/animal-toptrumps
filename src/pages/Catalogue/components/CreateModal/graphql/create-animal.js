import { gql } from "@apollo/client";

export const CREATE_ANIMAL = gql`
  mutation($values: AnimalInput!) {
    createAnimal(values: $values) {
      id
      name
      type
      diet
      extinct
    }
  }
`;
