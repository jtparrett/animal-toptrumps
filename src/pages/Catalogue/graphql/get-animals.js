import { gql } from "@apollo/client";

import { AnimalCardFragment } from "../../../components";

export const GET_ANIMALS = gql`
  ${AnimalCardFragment}

  {
    getAnimals {
      id
      ...AnimalCardFragment
    }
  }
`;
