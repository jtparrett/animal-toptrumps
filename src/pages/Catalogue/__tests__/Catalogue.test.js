import React from "react";
import { render, act, wait } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { ThemeProvider, theme } from "@chakra-ui/core";

import { Catalogue } from "../";
import { GET_ANIMALS } from "../graphql";
import { ANIMAL_TYPE, ANIMAL_DIET_TYPE } from "../../../consts";

const mocks = [
  {
    request: {
      query: GET_ANIMALS,
    },
    result: {
      data: {
        getAnimals: [
          {
            id: "0",
            name: "Animal 0",
            type: ANIMAL_TYPE.MAMMAL,
            diet: ANIMAL_DIET_TYPE.HERBIVORE,
            extinct: false,
            __typename: "Animal",
          },
          {
            id: "1",
            name: "Animal 1",
            type: ANIMAL_TYPE.AMPHIBIOUS,
            diet: ANIMAL_DIET_TYPE.HERBIVORE,
            extinct: true,
            __typename: "Animal",
          },
        ],
      },
    },
  },
];

describe("<Catalogue />", () => {
  it("renders", async () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <MockedProvider mocks={mocks}>
          <Catalogue />
        </MockedProvider>
      </ThemeProvider>
    );

    await act(async () => {
      await wait();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
