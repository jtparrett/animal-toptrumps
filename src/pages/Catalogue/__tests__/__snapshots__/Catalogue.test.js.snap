// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Catalogue /> renders 1`] = `
<DocumentFragment>
  <div
    class="css-f5mycz"
  >
    <div
      class="css-1uanfen"
    >
      <h1
        class="css-4kl02x"
      >
        Catalogue
      </h1>
      <button
        class="css-c4fa1y"
        type="button"
      >
        Create animal
        <svg
          aria-hidden="true"
          class="css-yyruks"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-2zdq89"
    >
      <article
        class="css-19xshe8"
      >
        <div
          class="css-1lm5fr4"
        >
          <h1
            class="css-gpthjz"
          >
            Animal 0
          </h1>
          <div
            class="css-wsi9t5-Skeleton"
          >
            <div
              class="css-emr9vq"
            >
              <p
                class="css-12kwuzl"
              >
                Type
              </p>
              <p
                class="css-1t24val"
              >
                Mammal
              </p>
              <svg
                class="css-1i71l27"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64.264 44.704c-88.765 25.213-39.73 158.676-2.108 161.887-8.506 42.218-13.32 84.645-12.642 127.824.355 22.67 51.374 58.424 105.35 86.035C186.48 466.787 238.98 466.876 256 467.296c17.02-.42 69.52-.51 101.137-46.846 53.975-27.61 104.994-63.364 105.35-86.035.677-43.179-4.137-85.606-12.643-127.824 37.623-3.211 86.657-136.674-2.108-161.887-28.433 1.462-55.606 16.152-82.172 37.223C317.64 62.233 293.164 61.296 256 61.296c-37.163 0-61.639.937-109.564 20.63-26.566-21.07-53.739-35.76-82.172-37.222zm4.927 49.928c6.057-.096 17.237 5.625 26.809 14.664 3.362 3.175-22.916 51.05-26.818 50.158-4.59-1.05-14.985-48.285-3.725-63.318.745-.994 2.039-1.477 3.734-1.504zm373.618 0c1.695.027 2.99.51 3.734 1.504 11.26 15.033.865 62.268-3.725 63.318-3.902.893-30.18-46.983-26.818-50.158 9.572-9.04 20.752-14.76 26.809-14.664zm-260.85 82.088c17.457-.304 30.686 24.445 42.041 44.576-53.57 15.878-82.575-17.11-57.752-37.777 5.603-4.665 10.823-6.714 15.711-6.8zm148.082 0c4.888.085 10.108 2.134 15.711 6.799 24.823 20.667-4.181 53.655-57.752 37.777 11.355-20.13 24.584-44.88 42.041-44.576zM256 251.548c17.32.223 36.826 9.567 30.197 36.517-1.444 5.872-12.516 8.617-21.197 9.7v8.972c99.884 41.126 79.565 138.559-9 138.559s-108.884-97.433-9-138.559v-8.972c-8.68-1.083-19.753-3.828-21.197-9.7-6.629-26.95 12.877-36.294 30.197-36.517z"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-wsi9t5-Skeleton"
          >
            <div
              class="css-emr9vq"
            >
              <p
                class="css-12kwuzl"
              >
                Diet type
              </p>
              <p
                class="css-1t24val"
              >
                Herbivore
              </p>
              <svg
                class="css-1i71l27"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478 23.74c-11.3.28-23.5 1.47-34.9 3.67-16 3.08-30.4 8.33-37.9 13.88-.7 2.21-.9 5.24-.8 7.29 0 2.1.3 4.38.7 6.68 13.3-9.91 29.6-15.59 46.1-20.96-14.3 10.07-26.6 20.65-38.2 31.49 8.1 3.57 17.7 3.62 23.9-.77h.1c13.4-9.64 31.3-25.76 41-41.25zm-297.9 3.59c-2.7 0-5.6.3-8.6.97-2 9.79-.8 18.7 2.9 25.79-9-3.54-19.8-2.95-30.8 6.26 6.7 22.83 26.9 33 47.7 22.69.6 9.44.9 18.96.9 28.46-6.5-7.1-17.2-13.55-28.3-13.25-7.8.3-15.9 3.85-23 12.85 11.2 21 33 26.8 51.2 12.6-.2 10.3-.7 20.6-1.5 31-6.1-8.4-17-16.4-29.1-16.3-5.4.1-11 1.8-16.6 5.8 16 4.5 30.6 10.9 43.7 19.5 4.9 3.1 7.4 7.9 8.8 12.6 1.3 4.7 1.6 9.6 1.5 14.6-.4 10.2-2.8 20.9-6.4 30l-1.3 3-2.9 1.5c-2 1.1-4.3 2.2-6.5 3.3-2.1 12.8-4.4 25.7-6.9 38.7-14.8-18-29.7-24.6-44.4-24.7-37.59-.4-74.36 41.2-108.68 41.9 46.47 3.3 118.78-44.1 143.98 11.6 18.7 13.9 37.9 25.2 57.5 33.8 1.6-9.4 4.2-19.3 7.8-28.1 2.3-5.6 4.9-10.8 8.2-15.2 3.3-4.4 7.3-8.4 13.4-9.9 1.7-.5 3.5-.9 5.2-1.3-5.4-10.5-4.7-24.6 1.7-36.4 6.5-12.1 18.3-20.5 30.4-21.4 12.4-18.9 25.4-37.3 39.1-55.4 10.6 22.5 34 26.6 53.3 10.9-4.9-18.1-18-23.8-30.5-23.5-5.6.1-11.1 1.5-15.8 3.5 19.7-25.5 40.8-50 63.4-73.76-8.2-3.62-8.8-6.35-10.9-14.61-23 23.99-44.5 48.97-64.6 74.67-.5 15.9-9.5 30.2-24.7 33.2h-.2C289.4 186.3 280 200 271 214c2.3-14.9 4.6-30.2 7.4-45.4-7-4.4-12.6-11.8-15.7-20.4-9.5 30.5-16.3 62.9-25.1 90.9-1.2-.2-2.5-.3-3.7-.3-15.8-.1-29.7 13.6-33.3 23.9 12.1 10.4 25.4 12.4 36.3 7.6-4.1 7.3-8.3 14.7-12.3 22.1-3.6-1.6-7.3-2.4-10.8-2.4-11.4-.1-21.8 7.1-28.1 14.9 6.9-32.8 13.1-65.6 17.6-98.3 18.4 9.3 37.5 1.1 46-19.6-7.2-7.5-14.8-10.5-22-10.6-7.9-.1-15.4 3.4-21.3 8.1 1.3-11.3 2.3-22.6 3-33.9 18.8 18.5 43.1 13.5 55.1-9.1-7.1-9-15.2-12.5-23-12.8-13.6-.4-26.4 9.3-31.9 18.1.6-10.7 1-21.3 1-31.9 20.5 9.6 40.2-.6 46.8-23.2-7.2-6.03-14.3-8.36-20.8-8.26-10.8.16-20.2 6.95-26.2 14.71-.2-7.05-.6-14.1-1.1-21.13 23.4 7.71 42.3-6.26 44.5-30.81-19.8-10.93-35-.74-42.9 11.84-1.8-14.09-11-30.7-30.4-30.72zm178 13.92c-36.8 5.82-59.7 28.28-75.7 58l2.4-.6c5.3-1 10.4-.5 15 1.2 12.2-23.81 30.2-44.39 58.3-58.6zM490.2 105c-47.4 67.9-110.8 96.8-173.1 128.2 5.3 10.5 4.6 24.3-1.7 36v.1c65.8-40.8 132.8-79.5 174.8-164.3zm-89.7 4.4c-15.8-.2-29.7 13.7-33.3 23.9 20.9 18 45.3 10.9 55.3-13.3-7.2-7.5-14.8-10.5-22-10.6zm-111 6.8c-.4.1-.8.1-1.2.2-6.3 1.2-12 10.1-9.7 21.9 2.3 11.8 11 18 17.2 16.7 6.3-1.2 12-10.1 9.7-21.9-2.2-11.1-9.8-17.2-16-16.9zM84.98 154.4c-1.49 0-3.08.1-4.59.1 9.39 25.3 19.49 46.9 40.81 59.9h.1c14.2 9 31.2 6.5 45.9.9l-42.4-35.1 52.9 28c1.5-8.9 6.7-20.8 1.3-29.3h-.1v-.1c-24.5-15.9-55.3-24.1-93.92-24.4zm206.92 81.3c-5.5.2-12.2 4-16.5 11.9-5.7 10.6-2.9 20.9 2.7 23.9 8.6 3.7 18.2-5 21.4-10.8 5.7-10.6 2.9-20.8-2.7-23.9-1.4-.8-3.1-1.1-4.9-1.1zm-3.1 54.3c-1.5 0-3.1 0-4.6.1-9.2.4-18.2 1.8-27 4.1h-.1c.1 0-1.5.7-3.4 3.3-1.9 2.6-4.1 6.5-5.9 11.2-2.7 6.5-4.8 14.3-6.2 21.6l64-10.1-51.9 20.6c5.4 8 6.9 11.3 8.2 17.2 8.3 5.3 17.1 7.7 28.2 5 27.6-6.5 49.3-29 72-55.5-26-11.9-50.2-17.7-73.3-17.5zm-137.2 17.9l-9 22.9c32.7 14.4 60.6 31.1 87.3 47.9l12.5-22.6c-31.9-9.8-62-26.3-90.8-48.2zm-12.6 41c-20.9 27.3-46.01 51-74.71 64.9l7.88 16.2c33.33-16.2 60.93-43.1 83.53-73.2-5.4-2.7-11-5.4-16.7-7.9zm20.3 9.7c-23.6 42.1-46.3 72.1-70.35 116.1l15.85 8.6c23.4-42.9 46-73 70.5-116.3-5.2-2.8-10.6-5.6-16-8.4zm26.2 14.2c-24.6 37.7-42.2 68.5-59.6 108.3l16.4 7.2c17.1-39 34.2-68.8 58.6-106.4-5.1-3.1-10.2-6.2-15.4-9.1zm23 13.7c-15.4 26-17.1 57-19.8 80.6l17.8 2.1c2.9-24.2 5.1-52.2 17.3-73.1z"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-wsi9t5-Skeleton"
          >
            <div
              class="css-emr9vq"
            >
              <p
                class="css-12kwuzl"
              >
                Extinct
              </p>
              <p
                class="css-1t24val"
              >
                Not extinct
              </p>
            </div>
          </div>
        </div>
        <div
          class="css-1ro9jph"
        >
          <div
            class="css-1ne8nzs"
          >
            <button
              class="css-t4ass8"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="css-yxiis9"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M19.452,7.5H4.547a.5.5,0,0,0-.5.545L5.334,22.181A2,2,0,0,0,7.326,24h9.347a2,2,0,0,0,1.992-1.819L19.95,8.045a.5.5,0,0,0-.129-.382A.5.5,0,0,0,19.452,7.5Zm-9.2,13a.75.75,0,0,1-1.5,0v-9a.75.75,0,0,1,1.5,0Zm5,0a.75.75,0,0,1-1.5,0v-9a.75.75,0,0,1,1.5,0Z"
                  />
                  <path
                    d="M22,4H17.25A.25.25,0,0,1,17,3.75V2.5A2.5,2.5,0,0,0,14.5,0h-5A2.5,2.5,0,0,0,7,2.5V3.75A.25.25,0,0,1,6.75,4H2A1,1,0,0,0,2,6H22a1,1,0,0,0,0-2ZM9,3.75V2.5A.5.5,0,0,1,9.5,2h5a.5.5,0,0,1,.5.5V3.75a.25.25,0,0,1-.25.25H9.25A.25.25,0,0,1,9,3.75Z"
                  />
                </g>
              </svg>
              Delete
            </button>
          </div>
          <div
            class="css-1rr4qq7"
          >
            <button
              class="css-12l6wyb"
              type="button"
            >
              Edit
            </button>
          </div>
        </div>
      </article>
      <article
        class="css-19xshe8"
      >
        <div
          class="css-1lm5fr4"
        >
          <h1
            class="css-gpthjz"
          >
            Animal 1
          </h1>
          <div
            class="css-wsi9t5-Skeleton"
          >
            <div
              class="css-emr9vq"
            >
              <p
                class="css-12kwuzl"
              >
                Type
              </p>
              <p
                class="css-1t24val"
              >
                Amphibious
              </p>
              <svg
                class="css-1i71l27"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M335.7 88.94c-4.742.194-9.563 1.486-14.204 4.165-38.934 22.48-89.77 21.953-127.79.002-6.09-3.516-12.285-4.61-18.145-3.892 5.914 7.778 9.438 17.572 9.438 28.09 0 23.15-17.037 42.83-39.176 45.095-12.775 14.92-21.553 31.807-24.386 49.983 44.73-23.79 90.947-35.572 137.064-35.508 46.15.064 92.197 11.987 136.56 35.62-2.69-18.15-11.216-35.043-23.794-49.92-.585.026-1.17.048-1.76.048-24.18 0-43.447-20.7-43.447-45.318 0-10.64 3.6-20.543 9.64-28.364zm-194.15 3.216c-12.67 0-23.277 10.85-23.277 25.15 0 14.297 10.608 25.147 23.278 25.147 12.67 0 23.276-10.85 23.276-25.148s-10.606-25.15-23.275-25.15zm227.956 0c-12.67 0-23.277 10.85-23.277 25.15 0 14.297 10.607 25.147 23.276 25.147 12.67 0 23.277-10.85 23.277-25.148s-10.608-25.15-23.277-25.15zm67.572 93.367c-8.525.088-17.893 1.546-27.853 4.243 6.926 19.457 8.57 40.725 2.695 62.656-4.26 15.896.933 37.475 11.7 54.758l4.69 7.53-7.02 5.43c-19.765 15.28-36.44 25.107-46.104 35.264-9.664 10.158-13.887 19.59-10.915 40.875l1.525 10.91c3.596 4.7 7.678 9.43 12.142 14.06 19.876-14.55 36.01-23.887 68.344-4.094-6.738-18.804 15.938-29.762 46.72-29.78-36.91-15.88-64.98-25.62-86.438-30.376 67.492-72.188 97.182-127.96 66-159.188-8.172-8.183-19.356-12.034-33.28-12.28-.73-.014-1.463-.016-2.204-.01zm-361.617.002c-.806-.01-1.606-.008-2.397.006-13.925.248-25.14 4.1-33.313 12.282-31.182 31.227-1.492 87 66 159.188-21.456 4.756-49.528 14.497-86.438 30.375 30.782.02 53.458 10.977 46.72 29.78 32.332-19.792 48.468-10.454 68.343 4.095 6.713-6.962 12.572-14.146 17.188-21.12l.537-3.85c2.972-21.283-1.25-30.716-10.914-40.874-9.664-10.157-26.34-19.984-46.106-35.265l-7.02-5.427 4.692-7.53c10.73-17.228 15.858-39.233 11.7-54.76-5.782-21.572-4.185-42.44 2.536-61.56-11.336-3.388-21.954-5.216-31.527-5.338zm183.038 9.66c-46.096-.065-92.3 12.827-137.574 38.846.47 4.387 1.292 8.825 2.494 13.31v.002c5.453 20.354.593 42.93-9.484 62.297 15.89 11.634 30.343 20.526 41.478 32.23 10.36 10.89 16.795 25.132 16.955 43.712-1.096 16.308-9.157 39.273-22.347 59.244 24.59-14.237 42.134-15.333 45.29 3.492 14.097-17.783 25.698-20.386 38.985-8.035-3.745-31.452-11.117-52.887-17.258-65.097-14.896-36.567-42.816-61.484-73.742-83.424l11.36-16.014c38.788 27.517 76.798 62.663 89.124 119.566 9.628.705 19.25.65 28.85-.16 12.362-56.81 50.334-91.918 89.085-119.408l11.36 16.016c-31.19 22.127-59.333 47.28-74.13 84.363-6.045 12.357-13.14 33.493-16.793 64.158 13.29-12.35 24.89-9.748 38.987 8.035 3.153-18.825 20.697-17.73 45.288-3.492-13.51-20.455-21.645-44.058-22.42-60.424.415-18.01 6.81-31.872 16.95-42.533 11.135-11.705 25.586-20.595 41.474-32.23-10.064-19.29-14.99-41.736-9.48-62.302 1.198-4.467 2.028-8.89 2.51-13.266-44.85-25.79-90.852-38.82-136.964-38.886z"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-wsi9t5-Skeleton"
          >
            <div
              class="css-emr9vq"
            >
              <p
                class="css-12kwuzl"
              >
                Diet type
              </p>
              <p
                class="css-1t24val"
              >
                Herbivore
              </p>
              <svg
                class="css-1i71l27"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478 23.74c-11.3.28-23.5 1.47-34.9 3.67-16 3.08-30.4 8.33-37.9 13.88-.7 2.21-.9 5.24-.8 7.29 0 2.1.3 4.38.7 6.68 13.3-9.91 29.6-15.59 46.1-20.96-14.3 10.07-26.6 20.65-38.2 31.49 8.1 3.57 17.7 3.62 23.9-.77h.1c13.4-9.64 31.3-25.76 41-41.25zm-297.9 3.59c-2.7 0-5.6.3-8.6.97-2 9.79-.8 18.7 2.9 25.79-9-3.54-19.8-2.95-30.8 6.26 6.7 22.83 26.9 33 47.7 22.69.6 9.44.9 18.96.9 28.46-6.5-7.1-17.2-13.55-28.3-13.25-7.8.3-15.9 3.85-23 12.85 11.2 21 33 26.8 51.2 12.6-.2 10.3-.7 20.6-1.5 31-6.1-8.4-17-16.4-29.1-16.3-5.4.1-11 1.8-16.6 5.8 16 4.5 30.6 10.9 43.7 19.5 4.9 3.1 7.4 7.9 8.8 12.6 1.3 4.7 1.6 9.6 1.5 14.6-.4 10.2-2.8 20.9-6.4 30l-1.3 3-2.9 1.5c-2 1.1-4.3 2.2-6.5 3.3-2.1 12.8-4.4 25.7-6.9 38.7-14.8-18-29.7-24.6-44.4-24.7-37.59-.4-74.36 41.2-108.68 41.9 46.47 3.3 118.78-44.1 143.98 11.6 18.7 13.9 37.9 25.2 57.5 33.8 1.6-9.4 4.2-19.3 7.8-28.1 2.3-5.6 4.9-10.8 8.2-15.2 3.3-4.4 7.3-8.4 13.4-9.9 1.7-.5 3.5-.9 5.2-1.3-5.4-10.5-4.7-24.6 1.7-36.4 6.5-12.1 18.3-20.5 30.4-21.4 12.4-18.9 25.4-37.3 39.1-55.4 10.6 22.5 34 26.6 53.3 10.9-4.9-18.1-18-23.8-30.5-23.5-5.6.1-11.1 1.5-15.8 3.5 19.7-25.5 40.8-50 63.4-73.76-8.2-3.62-8.8-6.35-10.9-14.61-23 23.99-44.5 48.97-64.6 74.67-.5 15.9-9.5 30.2-24.7 33.2h-.2C289.4 186.3 280 200 271 214c2.3-14.9 4.6-30.2 7.4-45.4-7-4.4-12.6-11.8-15.7-20.4-9.5 30.5-16.3 62.9-25.1 90.9-1.2-.2-2.5-.3-3.7-.3-15.8-.1-29.7 13.6-33.3 23.9 12.1 10.4 25.4 12.4 36.3 7.6-4.1 7.3-8.3 14.7-12.3 22.1-3.6-1.6-7.3-2.4-10.8-2.4-11.4-.1-21.8 7.1-28.1 14.9 6.9-32.8 13.1-65.6 17.6-98.3 18.4 9.3 37.5 1.1 46-19.6-7.2-7.5-14.8-10.5-22-10.6-7.9-.1-15.4 3.4-21.3 8.1 1.3-11.3 2.3-22.6 3-33.9 18.8 18.5 43.1 13.5 55.1-9.1-7.1-9-15.2-12.5-23-12.8-13.6-.4-26.4 9.3-31.9 18.1.6-10.7 1-21.3 1-31.9 20.5 9.6 40.2-.6 46.8-23.2-7.2-6.03-14.3-8.36-20.8-8.26-10.8.16-20.2 6.95-26.2 14.71-.2-7.05-.6-14.1-1.1-21.13 23.4 7.71 42.3-6.26 44.5-30.81-19.8-10.93-35-.74-42.9 11.84-1.8-14.09-11-30.7-30.4-30.72zm178 13.92c-36.8 5.82-59.7 28.28-75.7 58l2.4-.6c5.3-1 10.4-.5 15 1.2 12.2-23.81 30.2-44.39 58.3-58.6zM490.2 105c-47.4 67.9-110.8 96.8-173.1 128.2 5.3 10.5 4.6 24.3-1.7 36v.1c65.8-40.8 132.8-79.5 174.8-164.3zm-89.7 4.4c-15.8-.2-29.7 13.7-33.3 23.9 20.9 18 45.3 10.9 55.3-13.3-7.2-7.5-14.8-10.5-22-10.6zm-111 6.8c-.4.1-.8.1-1.2.2-6.3 1.2-12 10.1-9.7 21.9 2.3 11.8 11 18 17.2 16.7 6.3-1.2 12-10.1 9.7-21.9-2.2-11.1-9.8-17.2-16-16.9zM84.98 154.4c-1.49 0-3.08.1-4.59.1 9.39 25.3 19.49 46.9 40.81 59.9h.1c14.2 9 31.2 6.5 45.9.9l-42.4-35.1 52.9 28c1.5-8.9 6.7-20.8 1.3-29.3h-.1v-.1c-24.5-15.9-55.3-24.1-93.92-24.4zm206.92 81.3c-5.5.2-12.2 4-16.5 11.9-5.7 10.6-2.9 20.9 2.7 23.9 8.6 3.7 18.2-5 21.4-10.8 5.7-10.6 2.9-20.8-2.7-23.9-1.4-.8-3.1-1.1-4.9-1.1zm-3.1 54.3c-1.5 0-3.1 0-4.6.1-9.2.4-18.2 1.8-27 4.1h-.1c.1 0-1.5.7-3.4 3.3-1.9 2.6-4.1 6.5-5.9 11.2-2.7 6.5-4.8 14.3-6.2 21.6l64-10.1-51.9 20.6c5.4 8 6.9 11.3 8.2 17.2 8.3 5.3 17.1 7.7 28.2 5 27.6-6.5 49.3-29 72-55.5-26-11.9-50.2-17.7-73.3-17.5zm-137.2 17.9l-9 22.9c32.7 14.4 60.6 31.1 87.3 47.9l12.5-22.6c-31.9-9.8-62-26.3-90.8-48.2zm-12.6 41c-20.9 27.3-46.01 51-74.71 64.9l7.88 16.2c33.33-16.2 60.93-43.1 83.53-73.2-5.4-2.7-11-5.4-16.7-7.9zm20.3 9.7c-23.6 42.1-46.3 72.1-70.35 116.1l15.85 8.6c23.4-42.9 46-73 70.5-116.3-5.2-2.8-10.6-5.6-16-8.4zm26.2 14.2c-24.6 37.7-42.2 68.5-59.6 108.3l16.4 7.2c17.1-39 34.2-68.8 58.6-106.4-5.1-3.1-10.2-6.2-15.4-9.1zm23 13.7c-15.4 26-17.1 57-19.8 80.6l17.8 2.1c2.9-24.2 5.1-52.2 17.3-73.1z"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-wsi9t5-Skeleton"
          >
            <div
              class="css-emr9vq"
            >
              <p
                class="css-12kwuzl"
              >
                Extinct
              </p>
              <p
                class="css-1t24val"
              >
                Extinct
              </p>
            </div>
          </div>
        </div>
        <div
          class="css-1ro9jph"
        >
          <div
            class="css-1ne8nzs"
          >
            <button
              class="css-t4ass8"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="css-yxiis9"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M19.452,7.5H4.547a.5.5,0,0,0-.5.545L5.334,22.181A2,2,0,0,0,7.326,24h9.347a2,2,0,0,0,1.992-1.819L19.95,8.045a.5.5,0,0,0-.129-.382A.5.5,0,0,0,19.452,7.5Zm-9.2,13a.75.75,0,0,1-1.5,0v-9a.75.75,0,0,1,1.5,0Zm5,0a.75.75,0,0,1-1.5,0v-9a.75.75,0,0,1,1.5,0Z"
                  />
                  <path
                    d="M22,4H17.25A.25.25,0,0,1,17,3.75V2.5A2.5,2.5,0,0,0,14.5,0h-5A2.5,2.5,0,0,0,7,2.5V3.75A.25.25,0,0,1,6.75,4H2A1,1,0,0,0,2,6H22a1,1,0,0,0,0-2ZM9,3.75V2.5A.5.5,0,0,1,9.5,2h5a.5.5,0,0,1,.5.5V3.75a.25.25,0,0,1-.25.25H9.25A.25.25,0,0,1,9,3.75Z"
                  />
                </g>
              </svg>
              Delete
            </button>
          </div>
          <div
            class="css-1rr4qq7"
          >
            <button
              class="css-12l6wyb"
              type="button"
            >
              Edit
            </button>
          </div>
        </div>
      </article>
    </div>
  </div>
</DocumentFragment>
`;
